frontend/controller/app/identity.js

```diff
@@ -0,0 +1,252 @@
+'use strict'
+
+import { GIErrorUIRuntimeError, L, LError, LTags } from '@common/common.js'
+import sbp from '@sbp/sbp'
+import { LOGIN, LOGIN_ERROR } from '~/frontend/utils/events.js'
+import { Secret } from '~/shared/domains/chelonia/Secret.js'
+import { boxKeyPair, buildRegisterSaltRequest, computeCAndHc, decryptContractSalt, hash, hashPassword, randomNonce } from '~/shared/zkpp.js'
+// Using relative path to crypto.js instead of ~-path to workaround some esbuild bug
+import { CURVE25519XSALSA20POLY1305, EDWARDS25519SHA512BATCH, deriveKeyFromPassword, serializeKey } from '../../../shared/domains/chelonia/crypto.js'
+import { handleFetchResult } from '../utils/misc.js'
+import { cloneDeep } from '@model/contracts/shared/giLodash.js'
+import { LOGIN_COMPLETE } from '../../utils/events.js'
+
+const loadState = async (identityContractID: string, password: ?string) => {
+  if (password) {
+    const stateKeyEncryptionKeyFn = (stateEncryptionKeyId, salt) => {
+      return deriveKeyFromPassword(CURVE25519XSALSA20POLY1305, password, salt + stateEncryptionKeyId)
+    }
+
+    const { encryptionParams, value: state } = await sbp('gi.db/settings/loadEncrypted', identityContractID, stateKeyEncryptionKeyFn)
+
+    if (state) {
+      const cheloniaState = state.cheloniaState
```

Comment:
Does `cheloniaState` have a copy of all of the contract states?

Does `state` also have a copy of all of the contract states?

Reply:
Yes and yes.

Reply:
(However, only the `cheloniaState` copy is to be used)

Reply:
The various states are now better explained.

Reply:
Could you add a comment stating that somewhere (maybe here) and also just some of the backstory on why this is the case? (This is similar to my other comment above (or below, depending on where you're reading this on Github)